packages/snaps-rpc-methods/src/restricted/dialog.test.ts

```diff
@@ -58,7 +62,64 @@ describe('implementation', () => {
     expect(hooks.showDialog).toHaveBeenCalledWith(
       'foo',
       DialogType.Alert,
-      panel([heading('foo'), text('bar')]),
+      'bar',
+      undefined,
+    );
+  });
+
+  it('gets the interface data if an interface ID is passed', async () => {
+    const content = text('foo');
+
+    const hooks = {
+      showDialog: jest.fn(),
+      isOnPhishingList: jest.fn(),
+      maybeUpdatePhishingList: jest.fn(),
+      getInterface: jest.fn().mockReturnValue({ content, state: {} }),
+      createInterface: jest.fn().mockReturnValue('bar'),
+    };
+
+    const implementation = getDialogImplementation(hooks);
+
+    await implementation({
+      context: { origin: 'foo' },
+      method: 'snap_dialog',
+      params: {
+        type: 'alert',
+        id: 'bar',
+      },
+    });
+
+    expect(hooks.getInterface).toHaveBeenCalledWith('foo', 'bar');
+    expect(hooks.showDialog).toHaveBeenCalledTimes(1);
+    expect(hooks.showDialog).toHaveBeenCalledWith(
+      'foo',
+      DialogType.Alert,
+      'bar',
+      undefined,
+    );
+  });
+
+  it('creates a new interface if some content is passed', async () => {
+    const hooks = getMockDialogHooks();
+    const implementation = getDialogImplementation(hooks);
+
+    const content = panel([heading('foo'), text('bar')]);
+
+    await implementation({
+      context: { origin: 'foo' },
+      method: 'snap_dialog',
+      params: {
+        type: 'alert',
+        content,
+      },
+    });
+
+    expect(hooks.createInterface).toHaveBeenCalledWith('foo', content);
```

Comment:
Can we assert that `isOnPhishingList` is called?

Reply:
It's done in a test further down in the file